packages/twenty-server/src/modules/webhook/jobs/call-webhook.job.ts

```diff
@@ -34,10 +47,38 @@ export class CallWebhookJob {
     };
 
     try {
+      const headers: Record<string, string> = {
+        'Content-Type': 'application/json',
+      };
+
+      const {
+        record: { secret, ...recordWithoutSecret } = {},
+        ...dataWithoutRecord
+      } = data;
+
+      const payloadWithoutSecret = {
+        ...dataWithoutRecord,
+        record: recordWithoutSecret,
+      };
+
+      if (secret) {
```

Comment:
Oh ok no you were just trying to fetch secret this way. I don't think it works then? Try to start the worker with `npx nx worker twenty-server` and add a requestbin url or something where you can track the webhook. You'll see that it isn't encrypted. You need to get the secret at the same time you get the targetUrl and pass it down

Reply:
Initially, I assumed that the record object in the webhook event payload contains the secret that the user set for the webhook. This led me to destructure and remove the secret from record.

I only now realized that the record object only holds information about the updated or relevant record, not the actual secret for the webhook. And that the secret for signing the webhook can be fetched or derived in the CallWebhookJob logic (call-webhook-jobs.job.ts).

However, if I derive the secret in CallWebhookJob, then the very act of creating or updating the webhook (where the user sets the secret) might trigger a webhook event. This event would include the new webhook's configuration, potentially exposing the secret.

To prevent this, I’ll sanitize the record for events of type webhook to exclude the secret after making the above changes.

Reply:
Great! For the part about removing the secret, we try to avoid adding comments to the code so in that case creating a function with an explicit name could help for clarity